### PR TITLE
Remove English language + Refactor for formdata_to_model

### LINKED ISSUE #2103: workflows: 'en' language should generally not be written in the record
Currently, on the submission form, `en` is the default language, so
```json
"languages": ["en"]
```
gets written in the record. We don't want this, as the overwhelming majority of records are in English.
Instead, at some point in the submission/workflow, the following should happen:
if there is no `title_translations` or `languages` contains only `en`, remove languages.

### PR DESCRIPTION
**Pull request by rikirenz**
This PR wants to close #2103 #2000.

### CODE REVIEW

File: inspirehep/modules/literaturesuggest/form_normalization.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+from sqlalchemy.orm.exc import NoResultFound
+
+from invenio_accounts.models import User
+from invenio_oauthclient.models import UserIdentity
+
+from inspirehep.utils.pubnote import split_page_artid
+from inspirehep.utils.normalizers import normalize_journal_title
+
+
+def normalization_formdata(workflow_object, formdata):
```
Comment (jacquerie) [inline inspirehep/modules/literaturesuggest/form_normalization.py]: `s/normalization/normalize`: function names in general should be imperative (a cool read on the topic: http://steve-yegge.blogspot.ch/2006/03/execution-in-kingdom-of-nouns.html)

Comment (jacquerie) [inline inspirehep/modules/literaturesuggest/form_normalization.py]: I would also argue that a `workflow_object` is consistently called `obj` in the rest of the codebase, so it's better to call it that. But I'd like @david-caro's opinion on this.

File: inspirehep/modules/literaturesuggest/form_normalization.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2017 CERN.
```
Comment (jacquerie) [inline inspirehep/modules/literaturesuggest/form_normalization.py]: You'll need to run `scripts/prettify_licenses`

File: inspirehep/modules/literaturesuggest/form_normalization.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+def journal_title_normalization(data):
+    if data.get('journal_title'):
+        data['journal_title'] = normalize_journal_title(
+            data['journal_title']
+        )
+    return data
+
+
+def get_user_mail(workflow_obj, data):
```
Comment (jacquerie) [inline inspirehep/modules/literaturesuggest/form_normalization.py]: `s/data/formdata/`: (You might enjoy this clickbaity article on calling variables `data`: http://archive.oreilly.com/pub/post/the_worlds_two_worst_variable.html)

File: inspirehep/modules/literaturesuggest/form_normalization.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            id_user=workflow_obj.id_user,
+            method='orcid'
+        ).one().id
+    except NoResultFound:
+        data['orcid'] = None
+
+    return data
+
+
+def remove_english_language(data):
```
Comment (jacquerie) [inline inspirehep/modules/literaturesuggest/form_normalization.py]: It's weird that we populate this in the form, but then we remove it later. Of course this works, but can't we fix it in the form instead?

File: inspirehep/modules/literaturesuggest/form_normalization.py (hunk 1 above)
Comment (rikirenz) [inline inspirehep/modules/literaturesuggest/form_normalization.py]: IMHO I think `obj` is too much generic, but if we have a standard about that it is fine.

File: inspirehep/modules/literaturesuggest/form_normalization.py (hunk 4 above)
Comment (rikirenz) [inline inspirehep/modules/literaturesuggest/form_normalization.py]: We have already discuss about that. It should be fine.

File: inspirehep/modules/literaturesuggest/form_normalization.py (hunk 3 above)
Comment (rikirenz) [inline inspirehep/modules/literaturesuggest/form_normalization.py]: like `obj`

**Pull request closed without merging**